File: app/src/main/java/com/example/ui/screens/IntelligenceFeedScreen.kt
package com.example.ui.screens

import androidx.compose.animation.*
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.viewmodel.MainViewModel
import kotlinx.coroutines.delay

@Composable
fun IntelligenceFeedScreen(
    viewModel: MainViewModel,
    onNavigateToJobs: () -> Unit,
    modifier: Modifier = Modifier
) {
    val selectedCountry by viewModel.selectedCountry.collectAsState()
    val allJobs by viewModel.allJobs.collectAsState()
    
    // Filtering indicators based on active country context
    val countryJobs = remember(allJobs, selectedCountry) {
        if (selectedCountry == "Worldwide") allJobs 
        else allJobs.filter { it.location.contains(selectedCountry, ignoreCase = true) }
    }

    var liveFeedCount by remember { mutableStateOf(128) }
    var activeCampaigns by remember { mutableStateOf(42) }
    
    // Simulate real-time stream counter increases
    LaunchedEffect(Unit) {
        while (true) {
            delay(4000)
            liveFeedCount += (1..3).random()
            if (Math.random() > 0.7) {
                activeCampaigns += (-1..1).random()
            }
        }
    }

    // Interactive custom simulated event logs dynamically using real jobs
    val customLogs = remember(countryJobs, selectedCountry) {
        val logs = mutableListOf<String>()
        if (countryJobs.isNotEmpty()) {
            countryJobs.take(3).forEach { job ->
                logs.add("💼 LIVE ROLE: ${job.company} is hiring a ${job.title} in ${job.location}.")
            }
            logs.add("🔋 TELEMETRY: Telecom recruiter activity surged by 12% in $selectedCountry region.")
            logs.add("🤖 AI ANALYSIS: Local sector seeks increased experience with Material 3 and Jetpack Compose.")
            logs.add("📈 TRENDING: Key competencies in $selectedCountry include Kotlin, Room, and Coroutines.")
        } else {
            logs.add("💼 INTELLIGENCE: Real-time telemetry feed active for $selectedCountry.")
            logs.add("🔋 TELEMETRY: System listening for incoming API jobs ingestion campaigns.")
            logs.add("🤖 AI ANALYSIS: Applicant matching engines analyzing local demand signals.")
        }
        logs
    }

    LazyColumn(
        modifier = modifier
            .fillMaxSize()
            .background(Color(0xFF0F172A)), // Deep Dark Slate
        contentPadding = PaddingValues(16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Hero / Header Card
        item {
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .border(1.dp, Color(0xFF334155).copy(alpha = 0.5f), RoundedCornerShape(16.dp)),
                colors = CardDefaults.cardColors(containerColor = Color(0xFF1E293B)),
                shape = RoundedCornerShape(16.dp)
            ) {
                Column(
                    modifier = Modifier.padding(18.dp)
                ) {
                    Row(
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Box(
                            modifier = Modifier
                                .size(8.dp)
                                .clip(RoundedCornerShape(100))
                                .background(Color(0xFF10B981)) // Glow Green
                        )
                        Spacer(modifier = Modifier.width(8.dp))
                        Text(
                            text = "REAL-TIME INTELLIGENCE MATRIX",
                            style = MaterialTheme.typography.labelMedium,
                            fontWeight = FontWeight.Bold,
                            color = Color(0xFF10B981),
                            fontFamily = FontFamily.Monospace,
                            letterSpacing = 1.sp
                        )
                    }
                    
                    Spacer(modifier = Modifier.height(8.dp))
                    
                    Text(
                        text = if (selectedCountry == "Worldwide") "Global Hiring Telemetry" else "Telemetry: $selectedCountry",
                        style = MaterialTheme.typography.headlineSmall,
                        fontWeight = FontWeight.Black,
                        color = Color.White
                    )
                    
                    Spacer(modifier = Modifier.height(4.dp))
                    
                    Text(
                        text = "Aggregating ingestion metrics, deduplication indices, and live job vacancy telemetry from top employers.",
                        style = MaterialTheme.typography.bodyMedium,
                        color = Color(0xFF94A3B8)
                    )
                }
            }
        }

        // Real-Time System Metrics GRID
        item {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                MetricCard(
                    title = "INGESTED VAULT",
                    value = "$liveFeedCount",
                    sub = "Real-time records",
                    color = Color(0xFF3B82F6), // Blue
                    modifier = Modifier.weight(1f)
                )
                MetricCard(
                    title = "ACTIVE RECRUITING",
                    value = "$activeCampaigns",
                    sub = "Live campaigns",
                    color = Color(0xFF8B5CF6), // Violet
                    modifier = Modifier.weight(1f)
                )
            }
        }

        // Live Feed Updates (Event logs)
        item {
            Text(
                text = "LIVE SYSTEM INGEST STREAM",
                style = MaterialTheme.typography.labelMedium,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF64748B),
                fontFamily = FontFamily.Monospace,
                modifier = Modifier.padding(start = 4.dp, top = 8.dp)
            )
        }

        items(customLogs) { log ->
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .border(1.dp, Color(0xFF334155).copy(alpha = 0.3f), RoundedCornerShape(12.dp)),
                colors = CardDefaults.cardColors(containerColor = Color(0xFF1E293B).copy(alpha = 0.8f)),
                shape = RoundedCornerShape(12.dp)
            ) {
                Row(
                    modifier = Modifier.padding(14.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Box(
                        modifier = Modifier
                            .size(6.dp)
                            .clip(RoundedCornerShape(100))
                            .background(Color(0xFF3B82F6))
                    )
                    Spacer(modifier = Modifier.width(12.dp))
                    Text(
                        text = log,
                        style = MaterialTheme.typography.bodyMedium,
                        color = Color(0xFFE2E8F0),
                        fontFamily = FontFamily.Monospace,
                        fontSize = 12.sp,
                        lineHeight = 16.sp
                    )
                }
            }
        }

        // Call To Action to View All Jobs
        item {
            Spacer(modifier = Modifier.height(8.dp))
            Button(
                onClick = onNavigateToJobs,
                colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF3B82F6)),
                shape = RoundedCornerShape(12.dp),
                modifier = Modifier
                    .fillMaxWidth()
                    .height(52.dp)
                    .testTag("intel_view_jobs_btn")
            ) {
                Icon(Icons.Default.TrendingUp, contentDescription = null)
                Spacer(modifier = Modifier.width(8.dp))
                Text(
                    text = "EXPLORE ACTIVE FEED (${countryJobs.size} JOBS AVAILABLE)",
                    fontWeight = FontWeight.Bold,
                    letterSpacing = 0.5.sp
                )
            }
        }
    }
}

@Composable
fun MetricCard(
    title: String,
    value: String,
    sub: String,
    color: Color,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.border(1.dp, Color(0xFF334155).copy(alpha = 0.5f), RoundedCornerShape(12.dp)),
        colors = CardDefaults.cardColors(containerColor = Color(0xFF1E293B)),
        shape = RoundedCornerShape(12.dp)
    ) {
        Column(
            modifier = Modifier.padding(16.dp)
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.labelSmall,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF94A3B8),
                fontFamily = FontFamily.Monospace
            )
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = value,
                style = MaterialTheme.typography.headlineMedium,
                fontWeight = FontWeight.Black,
                color = color
            )
            Spacer(modifier = Modifier.height(2.dp))
            Text(
                text = sub,
                style = MaterialTheme.typography.bodySmall,
                color = Color(0xFF64748B)
            )
        }
    }
}

File: lib/screens/intelligence_feed_screen.dart
import 'dart:async';
import 'dart:math';
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../providers/jobs_provider.dart';

class IntelligenceFeedScreen extends StatefulWidget {
  final VoidCallback onNavigateToJobs;

  const IntelligenceFeedScreen({
    super.key,
    required this.onNavigateToJobs,
  });

  @override
  State<IntelligenceFeedScreen> createState() => _IntelligenceFeedScreenState();
}

class _IntelligenceFeedScreenState extends State<IntelligenceFeedScreen> {
  int _liveFeedCount = 128;
  int _activeCampaigns = 42;
  Timer? _simulationTimer;
  final Random _random = Random();

  @override
  void initState() {
    super.initState();
    _simulationTimer = Timer.periodic(const Duration(seconds: 4), (timer) {
      if (mounted) {
        setState(() {
          _liveFeedCount += _random.nextInt(3) + 1;
          if (_random.nextDouble() > 0.7) {
            _activeCampaigns += _random.nextInt(3) - 1;
          }
        });
      }
    });
  }

  @override
  void dispose() {
    _simulationTimer?.cancel();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final provider = Provider.of<JobsProvider>(context);
    final selectedCountry = provider.selectedCountry;
    final allJobs = provider.allJobs;

    // Filtering indicators based on active country context
    final countryJobs = selectedCountry == 'Worldwide'
        ? allJobs
        : allJobs.where((j) => j.location.toLowerCase().contains(selectedCountry.toLowerCase())).toList();

    // Interactive custom simulated event logs dynamically using real jobs
    final List<String> customLogs = [];
    if (countryJobs.isNotEmpty) {
      countryJobs.take(3).forEach((job) {
        customLogs.add('💼 LIVE ROLE: ${job.company} is hiring a ${job.title} in ${job.location}.');
      });
      customLogs.add('🔋 TELEMETRY: Telecom recruiter activity surged by 12% in $selectedCountry region.');
      customLogs.add('🤖 AI ANALYSIS: Local sector seeks increased experience with Flutter & Material 3 design elements.');
      customLogs.add('📈 TRENDING: Key competencies in $selectedCountry include Dart, Provider, and Asynchronous streams.');
    } else {
      customLogs.add('💼 INTELLIGENCE: Real-time telemetry feed active for $selectedCountry.');
      customLogs.add('🔋 TELEMETRY: System listening for incoming API jobs ingestion campaigns.');
      customLogs.add('🤖 AI ANALYSIS: Applicant matching engines analyzing local demand signals.');
    }

    return Scaffold(
      backgroundColor: const Color(0xFF0F172A),
      appBar: AppBar(
        title: const Text('Intelligence Matrix', style: TextStyle(fontWeight: FontWeight.bold)),
      ),
      body: SafeArea(
        child: ListView(
          padding: const EdgeInsets.all(16.0),
          children: [
            // Hero / Header Card
            Card(
              color: const Color(0xFF1E293B),
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(16),
                side: BorderSide(color: const Color(0xFF334155).withOpacity(0.5)),
              ),
              child: Padding(
                padding: const EdgeInsets.all(18.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Row(
                      children: [
                        Container(
                          width: 8,
                          height: 8,
                          decoration: const BoxDecoration(color: Color(0xFF10B981), shape: BoxShape.circle),
                        ),
                        const SizedBox(width: 8),
                        const Text(
                          'REAL-TIME INTELLIGENCE MATRIX',
                          style: TextStyle(
                            fontFamily: 'monospace',
                            fontSize: 11,
                            fontWeight: FontWeight.bold,
                            color: Color(0xFF10B981),
                            letterSpacing: 1.0,
                          ),
                        ),
                      ],
                    ),
                    const SizedBox(height: 12),
                    Text(
                      selectedCountry == 'Worldwide' ? 'Global Hiring Telemetry' : 'Telemetry: $selectedCountry',
                      style: const TextStyle(color: Colors.white, fontSize: 20, fontWeight: FontWeight.black),
                    ),
                    const SizedBox(height: 6),
                    const Text(
                      'Aggregating ingestion metrics, deduplication indices, and live job vacancy telemetry from top employers.',
                      style: TextStyle(color: Color(0xFF94A3B8), fontSize: 13, height: 1.4),
                    ),
                  ],
                ),
              ),
            ),
            const SizedBox(height: 16),

            // Real-Time System Metrics GRID
            Row(
              children: [
                Expanded(
                  child: _buildMetricCard(
                    'INGESTED VAULT',
                    '$_liveFeedCount',
                    'Real-time records',
                    const Color(0xFF3B82F6),
                  ),
                ),
                const SizedBox(width: 12),
                Expanded(
                  child: _buildMetricCard(
                    'ACTIVE RECRUITING',
                    '$_activeCampaigns',
                    'Live campaigns',
                    const Color(0xFF8B5CF6),
                  ),
                ),
              ],
            ),
            const SizedBox(height: 24),

            // Live Feed Updates (Event logs)
            const Padding(
              padding: EdgeInsets.only(left: 4.0),
              child: Text(
                'LIVE SYSTEM INGEST STREAM',
                style: TextStyle(
                  fontFamily: 'monospace',
                  fontSize: 11,
                  fontWeight: FontWeight.bold,
                  color: Color(0xFF64748B),
                  letterSpacing: 1.0,
                ),
              ),
            ),
            const SizedBox(height: 12),

            ...customLogs.map((log) {
              return Padding(
                padding: const EdgeInsets.only(bottom: 12.0),
                child: Card(
                  color: const Color(0xFF1E293B).withOpacity(0.8),
                  shape: RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(12),
                    side: BorderSide(color: const Color(0xFF334155).withOpacity(0.3)),
                  ),
                  child: Padding(
                    padding: const EdgeInsets.all(14.0),
                    child: Row(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Padding(
                          padding: const EdgeInsets.only(top: 5.0),
                          child: Container(
                            width: 6,
                            height: 6,
                            decoration: const BoxDecoration(color: Color(0xFF3B82F6), shape: BoxShape.circle),
                          ),
                        ),
                        const SizedBox(width: 12),
                        Expanded(
                          child: Text(
                            log,
                            style: const TextStyle(
                              color: Color(0xFFE2E8F0),
                              fontFamily: 'monospace',
                              fontSize: 12,
                              height: 1.4,
                            ),
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
              );
            }).toList(),

            const SizedBox(height: 16),

            // Call To Action to View All Jobs
            SizedBox(
              height: 52,
              child: ElevatedButton.icon(
                onPressed: widget.onNavigateToJobs,
                icon: const Icon(Icons.trending_up, color: Colors.white),
                label: Text(
                  'EXPLORE ACTIVE FEED (${countryJobs.length} JOBS AVAILABLE)',
                  style: const TextStyle(color: Colors.white, fontWeight: FontWeight.bold, letterSpacing: 0.5),
                ),
                style: ElevatedButton.styleFrom(
                  backgroundColor: const Color(0xFF3B82F6),
                  shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildMetricCard(String title, String value, String sub, Color tintColor) {
    return Card(
      color: const Color(0xFF1E293B),
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(12),
        side: BorderSide(color: const Color(0xFF334155).withOpacity(0.5)),
      ),
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              title,
              style: const TextStyle(
                fontFamily: 'monospace',
                fontSize: 10,
                fontWeight: FontWeight.bold,
                color: Color(0xFF94A3B8),
              ),
            ),
            const SizedBox(height: 8),
            Text(
              value,
              style: TextStyle(color: tintColor, fontWeight: FontWeight.black, fontSize: 24),
            ),
            const SizedBox(height: 2),
            Text(
              sub,
              style: const TextStyle(color: Color(0xFF64748B), fontSize: 11),
            ),
          ],
        ),
      ),
    );
  }
}
